src/libraries/Microsoft.PowerFx.Core/Public/Config/PowerFxConfig.cs

```diff
@@ -42,6 +43,11 @@ public sealed class PowerFxConfig
 
         public int MaxCallDepth { get; set; }
 
+        /// <summary>
+        /// Used with recursive user-defined functions to break long running functions, a good example of this is fibonacci. 
+        /// </summary>
+        public int MaxOperationsAllowed { get; set; }

```

Comment:
>MaxOperationsAllowed [](http://example.com/codeflow?start=19&length=20)

how does this compare to MaxCallDepth?

Reply:
MaxCallDepth tells us how deep the recursive call is, in cases like Fibonacci recursive tree doesn't go that deep but it's wide and this can potentially block the app. To avoid this we use MaxOperationsAllowed .

Reply:
What *exactly* does that mean?  

For example, in C# interpreter we have a timeout that signals on a timeout - that accepts a TimeValue is a well defined quota.  But was is an "operation"?

Reply:
Could you add a test where you 
1/ set MaxOps = 10, 
2/ use an UDF that will make Ops=30 
3/ return an error "Too many operations" of some sort, in the middle of the execution
4/ check we have that error & message